src/dnslinkers/route53.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+// @ts-ignore
+
+const { Route53Client, ChangeResourceRecordSetsCommand } = require('@aws-sdk/client-route-53')
+const isEmpty = require('lodash.isempty')
+const TTL = 60
+
+/**
+ * @typedef {import('./types').DNSRecord} DNSRecord
+ * @typedef {import('./types').Route53Options} Route53Options
+ */
+
+class Route53 {
+  /**
+   * @param {Route53Options} options
+   */
+  constructor ({ accessKeyId, secretAccessKey, region, hostedZoneId, record }) {
+    if ([accessKeyId, secretAccessKey, region, hostedZoneId, record].every(isEmpty)) {
+      throw new Error('accessKeyId, secretAccessKey, region, hostedZoneId and record are required for Route53')
+    }
+
+    this.accessKeyId = accessKeyId
+    this.secretAccessKey = secretAccessKey
+    this.region = region
+    this.hostedZoneId = hostedZoneId
+    this.record = record
+  }
+
+  /**
+   * @param {string} cid
+   * @returns {Promise<DNSRecord>}
+   */
+  async link (cid) {
+    const credentials = {
+      accessKeyId: this.accessKeyId,
+      secretAccessKey: this.secretAccessKey
+    }
+
+    const txtValue = `dnslink=/ipfs/${cid}`
+    const client = new Route53Client({ region: this.region, credentials })
+    const command = new ChangeResourceRecordSetsCommand({
+      HostedZoneId: this.hostedZoneId,
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: 'UPSERT',
+            ResourceRecordSet: {
+              Name: this.record,
+              Type: 'TXT',
+              TTL: TTL,
+              ResourceRecords: [
+                {
+                  Value: `"${txtValue}"`
+                }
+              ]
+            }
+          }
+        ]
+      }
+    })
+
+    try {
+
+   await client.send(command)
+
+  } catch (err) {
+      console.log(err)
```

Comment:
We should just let it throw the error. It'll be handled by the caller.

Reply:
ah ok let me update that thanks!